### DISCUSSION TITLE
Prisma relations not updating related objects

### DISCUSSION THREAD

**Discussion by wassimbenr** [prisma/prisma#19070] [Q&A] [answered]
I'm building a Next.js and Nest.js project with Prisma as my ORM, and I'm encountering an issue with the way Prisma manages relationships between my User and Project models. Specifically, when I create a new Project and associate it with one or more User objects, the Project is successfully created with the correct user data, but the corresponding User objects do not appear to be updated with the new project information in their projects array.

Here's an excerpt of my Prisma schema for reference:

```prisma
model User {
  id                String         @id @default(auto()) @map("_id") @db.ObjectId
  //...
  projects          Project[]      @relation(name: "users", fields: [project_ids], references: [id])
  project_ids       String[]       @db.ObjectId
  createdProjects   Project[]      @relation(name: "pmId")
  //...

  @@map(name: "users")
}

model Project {
  id             String        @id @default(auto()) @map("_id") @db.ObjectId
  //...
  users          User[]        @relation(name: "users", fields: [user_ids], references: [id])
  user_ids       String[]      @db.ObjectId
  pm             User?         @relation(name: "pmId", fields: [pmId], references: [id], onDelete: Cascade)
  pmId           String?       @db.ObjectId
  //...
  @@map(name: "projects")
}
```
Service:
```
create(data: Prisma.ProjectCreateInput) {
    return this.prisma.project.create({ data });
  }
```
Controller:
```
@Post()
  @ApiCreatedResponse({ type: ProjectEntity })
  create(@Body() data: Prisma.ProjectCreateInput) {
    return this.projectService.create(data);
  }
```

However, when I check the projects array of each User object, it doesn't seem to have been updated with the new project information. I've tried various debugging steps, including checking my Prisma schema, verifying my database backend (MongoDB) configuration, and using Prisma's debugging features, but I can't seem to identify the issue.

Any help or guidance on how to resolve this issue would be greatly appreciated. Thanks in advance!

**Comment by ludralph** [comment]
Hi @Dedpy 👋 

Can you please share the query you are performing?

**Reply by wassimbenr** [reply]
Hi @ludralph , thanks for the reply.
I've added the method call, it's just a simple create method

**Reply by ludralph** [reply]
Hi @Dedpy, What's the shape of the `data` that you are passing to the create method?

**Reply by wassimbenr** [reply]
```
{
    "title": "Project 1",
    "description":"This project Description",
    "organizationId": "6424a1a8bb3e6218df3f902d",
    "pmId": "64049614e57a421e00fdec2a",
    "user_ids": [
        "6408ebd780812e4415d8f197",
        "640926c17012cb0bce1bb746"
    ]
}
```

**Reply by ludralph** [reply]
You will need to update the projects array on each associated user by adding the newly created project to the array. You can leverage `update` method and `connect` to do this. See this section of the [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#:~:text=You%20can%20connect%20%20an%20existing%20record%20to%20a%20new%20or%20existing%20user.%20The%20following%20query%3A)

**Reply by wassimbenr** [reply]
I am accustomed to having the update occur automatically with Prisma when I pass an array of related object IDs. However, in this specific case, I am unsure why it is not functioning properly. I have other entities that are linked to user objects, and when I execute a POST method and include an array of IDs, the connection is already established with the user object.

**Reply by ludralph** [reply]
Hi @Dedpy 👋 

Can you try this script which you can use to connect Projects with Users and it would show connected projects in users as well.
```js
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {

  await prisma.user.deleteMany()
  await prisma.project.deleteMany()
  
  await prisma.user.create({
    data:{
      id:'6408ebd780812e4415d8f197',
    }
  })

  await prisma.user.create({
    data:{
      id:'640926c17012cb0bce1bb746',
    }
  })

  await prisma.project.create({
    data:{
      users:{
        connect:[
          {
            id:'6408ebd780812e4415d8f197'
          },
          {
            id:'640926c17012cb0bce1bb746'
          }
        ]
      }
    }
  })
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```
![Screenshot 2023-05-04 at 7 23 41 PM](https://user-images.githubusercontent.com/13855608/236232181-16647e17-843c-4165-8e9f-4d5f6bbfd7cb.png)

**Reply by wassimbenr** [reply, marked as answer]
@ludralph Thank you very much for offering your help; it is much appreciated. 

I am trying to create a project where I need to connect it with existing users by passing their IDs in the data. 

However, when I attempt to remove the `Prisma.ProjectCreateInput` and replace it with `any`, it worked perfectly.

While this approach worked, I found that ProjectCreateInput does not include the user_ids array. I also attempted to use `Prisma.ProjectUncheckedCreateInput`, but it resulted in errors when using the map functions.

This is what I went with as a solution: 

```
 createProjectWithUsers(data: any) {
    return this.prisma.project.create({
      data: {
        ...data,
        users: {
          connect: data.user_ids.map((id: string) => ({ id })),
        },
  }
```
